test/e2e/cache/create_enable_add_with_private_registry_delete.go

```diff
@@ -0,0 +1,283 @@
+		framework.ExpectNoError(err)
+		rootPodExecutor := framework.NewRootPodExecutor(f.Logger, f.ShootFramework.ShootClient, &nodeList.Items[0].Name, "kube-system")
+
+		pullImage := fmt.Sprintf("ctr content fetch --all-platforms docker.io/library/%s > /dev/null", "nginx:1.24.0")
+		executeCommand(ctx, rootPodExecutor, pullImage)
+		tagImage := fmt.Sprintf("ctr images tag docker.io/library/%[1]s %[2]s/%[1]s > /dev/null", "nginx:1.24.0", upstreamHostPort)
+		executeCommand(ctx, rootPodExecutor, tagImage)
+		pushImage := fmt.Sprintf("ctr images push --plain-http -u admin:%s %s/%s > /dev/null", pass, upstreamHostPort, "nginx:1.24.0")
+		executeCommand(ctx, rootPodExecutor, pushImage)
+		rmImage := fmt.Sprintf("ctr images rm %s/%s > /dev/null", upstreamHostPort, "nginx:1.24.0")
+		executeCommand(ctx, rootPodExecutor, rmImage)
+		rmImage = fmt.Sprintf("ctr images rm %s/%s > /dev/null", "docker.io/library", "1.24.0")
+		executeCommand(ctx, rootPodExecutor, rmImage)
+
+		Expect(rootPodExecutor.Clean(ctx)).To(Succeed())
+
+		By("Enable the registry-cache extension")
+		ctx, cancel = context.WithTimeout(parentCtx, 10*time.Minute)
+		defer cancel()
+
+		Expect(f.UpdateShoot(ctx, f.Shoot, func(shoot *gardencorev1beta1.Shoot) error {
+			addPrivateRegistrySecret(shoot)
+			size := resource.MustParse("2Gi")
+			common.AddOrUpdateRegistryCacheExtension(shoot, []v1alpha3.RegistryCache{
+				{
+					Upstream: upstreamHostPort,
+					//RemoteURL:           ptr.To("http://" + upstreamHostPort), TODO
+					Volume:              &v1alpha3.Volume{Size: &size},
+					SecretReferenceName: ptr.To("upstream-secret"),
+				},
+			})
+
+			return nil
+		})).To(Succeed())
+
+		By("Wait until the registry configuration is applied")
+		ctx, cancel = context.WithTimeout(parentCtx, 5*time.Minute)
+		defer cancel()
+		common.WaitUntilRegistryCacheConfigurationsAreApplied(ctx, f.Logger, f.ShootFramework.ShootClient)
+
+		By("[" + upstreamHostPort + "] Verify registry-cache works")
+		common.VerifyRegistryCache(parentCtx, f.Logger, f.ShootFramework.ShootClient, upstreamHostPort, upstreamHostPort+"/nginx@sha256:3579c1362f12c540b4c48ffa4816348b0653044d1e1e728816f6f5c9e85774e3")
+
+		By("Delete Shoot")
+		ctx, cancel = context.WithTimeout(parentCtx, 10*time.Minute)
+		defer cancel()
+		Expect(f.DeleteShootAndWaitForDeletion(ctx, f.Shoot)).To(Succeed())
+	})
+})
+
+func addPrivateRegistrySecret(shoot *gardencorev1beta1.Shoot) {
+	shoot.Spec.Resources = append(shoot.Spec.Resources, gardencorev1beta1.NamedResourceReference{
+		Name: "upstream-secret",
+		ResourceRef: autoscalingv1.CrossVersionObjectReference{
+			APIVersion: "v1",
+			Kind:       "Secret",
+			Name:       "ro-upstream-secret",
+		},
+	})
+}
+
+// executeCommand executes a command on the host and checks the returned result
+func executeCommand(ctx context.Context, rootPodExecutor framework.RootPodExecutor, command string) {
+	response, err := rootPodExecutor.Execute(ctx, command)
+	framework.ExpectNoError(err)
```

Comment:
`ExpectWithOffset(1,` should be used instead as this is an helper func doing the expectations/assertions.